File: hydro/src/main/java/com/ultreon/hydro/event/SubscribeEvent.java
package com.ultreon.hydro.event;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface SubscribeEvent {
    boolean ignoreCancelled() default false;

    EventPriority priority() default EventPriority.NORMAL;
}

File: src/main/kotlin/qtech/bubbles/event/SubscribeEvent.kt
package qtech.bubbles.event

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class SubscribeEvent(val ignoreCancelled: Boolean = false, val priority: EventPriority = EventPriority.NORMAL)